File: android/src/main/java/me/timeto/app/ui/symbol/SymbolView.kt
package me.timeto.app.ui.symbol

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import me.timeto.shared.Symbol

@Composable
fun SymbolView(
    symbol: Symbol,
    color: Color,
    letterSize: TextUnit,
    iconSize: Dp,
    emojiSize: TextUnit,
    modifier: Modifier,
) {
    when (symbol) {
        is Symbol.Letter -> {
            SymbolLetterView(
                letter = symbol,
                color = color,
                size = letterSize,
                modifier = modifier,
            )
        }
        is Symbol.Icon -> {
            SymbolIconView(
                icon = symbol,
                color = color,
                size = iconSize,
                modifier = modifier,
            )
        }
        is Symbol.Emoji -> {
            SymbolEmojiView(
                emoji = symbol,
                size = emojiSize,
                modifier = modifier,
            )
        }
    }
}

File: apple/Common/Symbol/SymbolView.swift
import SwiftUI
import shared

struct SymbolView: View {
    
    let symbol: Symbol
    let color: Color
    let letterSize: CGFloat
    let iconSize: CGFloat
    let emojiSize: CGFloat
    
    var body: some View {
        if let letter = symbol as? Symbol.Letter {
            SymbolLetterView(
                letter: letter,
                color: color,
                size: letterSize,
            )
        } else if let icon = symbol as? Symbol.Icon {
            SymbolIconView(
                icon: icon,
                color: color,
                size: iconSize,
            )
        } else if let emoji = symbol as? Symbol.Emoji {
            SymbolEmojiView(
                emoji: emoji,
                size: emojiSize,
            )
        } else {
            // todo handle sealed like enum
        }
    }
}
